src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+
+        This method initializes and configures subcommands for the OTX CLI parser.
+        It iterates over the available subcommands, adds arguments specific to each subcommand,
+        and registers them with the parser.
+
+        Returns:
+            None
+        """
+        self._subcommand_parsers: dict[str, ArgumentParser] = {}
+        parser_subcommands = self.parser.add_subcommands()
+        self._set_extension_subcommands_parser(parser_subcommands)
+        if not _ENGINE_AVAILABLE:
+            return
+        for subcommand in self.engine_subcommands():
+            sub_parser = self.subcommand_parser()
+            engine_skip = {"model", "datamodule", "optimizer", "scheduler"}
+            sub_parser.add_class_arguments(
+                Engine,
+                "engine",
+                fail_untyped=False,
+                sub_configs=True,
+                instantiate=False,
+                skip=engine_skip,
+            )
+            sub_parser.link_arguments("data_root", "engine.data_root")
+
+            # Model Settings
+            from otx.core.model.entity.base import OTXModel
+
+            model_kwargs: dict[str, Any] = {"fail_untyped": False}
+
+            sub_parser.add_subclass_arguments(
+                OTXModel,
+                "model",
+                required=False,
+                **model_kwargs,
+            )
+            # Datamodule Settings
+            from otx.core.data.module import OTXDataModule
+
+            sub_parser.add_class_arguments(
+                OTXDataModule,
+                "data",
+                fail_untyped=False,
+                sub_configs=True,
+            )
+            sub_parser.link_arguments("data_root", "data.config.data_root")
+
+            # Optimizer & Scheduler Settings
+            from lightning.pytorch.cli import LRSchedulerTypeTuple
+            from torch.optim import Optimizer
+
+            optim_kwargs = {"instantiate": False, "fail_untyped": False, "skip": {"params"}}
+            scheduler_kwargs = {"instantiate": False, "fail_untyped": False, "skip": {"optimizer"}}
+            sub_parser.add_subclass_arguments(
+                baseclass=(Optimizer,),
+                nested_key="optimizer",
+                **optim_kwargs,
+            )
+            sub_parser.add_subclass_arguments(
+                baseclass=LRSchedulerTypeTuple,
+                nested_key="scheduler",
+                **scheduler_kwargs,
+            )
+
+            skip: set[str | int] = set(self.engine_subcommands()[subcommand])
+            fn = getattr(Engine, subcommand)
+            description = get_short_docstring(fn)
+            added = sub_parser.add_method_arguments(
+                Engine,
+                subcommand,
+                skip=skip,
+                fail_untyped=False,
+            )
+
+            # Load default subcommand config file
+            default_config_file = Path(get_otx_root_path()) / "recipe" / "_base_" / f"{subcommand}.yaml"
```

Comment:
maybe `recipe` could be renamed to `recipes` as it contains multiple files?

Reply:
I was torn between `configs` and `recipe` and `recipes`, so I left with the old name `recipe`. What do you think best?